docs: readme rewrite (#1262)
interactions-py/interactions.py: README.md

## Key Features
Interactions.py offers a range of features for building Discord bots with Python, including:
- ✅ 100% coverage of the Discord API
- ✅ Dynamic cache with TTL support
- ✅ Modern and Pythonic API for easy interaction with Discord
- ✅ Proper rate-limit handling
- ✅ Feature parity with most other Discord API wrappers
- ✅ Fully automated command synchronisation

In addition to core functionality, interactions.py provides a range of optional extensions, allowing you to further customize your bot and add new features with ease.

## Extensibility

So the base library doesn't do what you want? No problem! With builtin extensions, you are able to extend the functionality of the library. And if none of those pique your interest, there are a myriad of other extension libraries available.

Just type `bot.load("extension")`

> <details>

Reviewer: This could just be me, but this `<details>` section feels very odd. I'm not sure whether it's the fact that it's collapsible, or that we're putting all information regarding native extensions of the library here, but it just doesn't seem right nor fitting to belong here. Maybe I'm viewing this from a bad angle -- but maybe we should put this somewhere else, or should we?

Willing to hear rationale on this

Author: It was used to prevent reader fatigue. Large blocks of text will "scare" people off from reading a document. Extensions are a low-priority part of the library for initial "sale" however they are still a key selling point. 

For someone higher level, such as us, it's unnecessary and just adds an extra step to reading - for a less experienced dev it is a valuable technique to keep them engaged 

By hiding it; by the point, someone reaches the part they will expand it - sunk cost kicks in and they keep reading.

> ![image](https://img.shields.io/pypi/dm/discord-py-slash-command.svg)

Reviewer: I feel like we should move the badges down to be underneath L10, but that's just my preference.

Author: Ooooh this one is contentious. I'm a badges at the top kinda man

Reviewer: This is usually my preferred style:

---

# interactions.py
![image](https://img.shields.io/pypi/dm/discord-py-slash-command.svg)

A highly extensible, easy to use, and feature complete framework for Discord.

Author: Mine

--------

![image](https://img.shields.io/pypi/dm/discord-py-slash-command.svg)
# interactions.py
A highly extensible, easy to use, and feature-complete framework for Discord.

![image](https://img.shields.io/pypi/pyversions/discord-py-interactions.svg)
![image](https://img.shields.io/pypi/v/discord-py-interactions.svg)
![image](https://readthedocs.org/projects/interactionspy/badge/?version=latest)
[![Code style: black](https://img.shields.io/badge/code%20style-black-000000.svg)](https://github.com/psf/black)
![image](https://discord.com/api/guilds/789032594456576001/embed.png)

------------------------------------------------------------------------

> # Interactions.py - A Feature-rich Discord Bot Framework for Python

Reviewer: ```suggestion
# interactions.py
A feature-rich Discord client library, written in Python
```

Author: Disagree on this. The tagline is valuable for both SEO, and for hooking a user with the cliff notes lib features

Reviewer: I think we should look into adding more tags if we're looking at it from an SEO standpoint

Reviewer: re: "library" vs. "framework" -- would this also be applicable here? I'm also more for the suggested change than the current proposal, FWIW

Author: Yes

Reviewer: Resolving this. As much as I dislike it, I can't really go against potential SEO gains

> A highly extensible, easy to use, and feature complete bot framework for Discord, `Interactions.py` is the culmination of years of experience in the Discord API and bot development. This library has been built from the ground up with community feedback and suggestions in mind, providing a modern and intuitive API for easy interaction with the Discord API in Python.

Reviewer: ```suggestion
A highly extensible, easy to use, and feature complete framework for Discord.

interactions.py is the culmination of years of experience with Discord's APIs and bot development. This framework has been built from the ground up with community feedback and suggestions in mind. Our framework provides a modern and intuitive set of language bindings for easy interaction with Discord.
```

Author: Bot framework was used to match venacular used by other libraries. Happy to drop the "bot" but I think "framework" should remain

## Key Features

> Interactions.py offers a range of features for building Discord bots with Python, including:

Reviewer: ```suggestion
interactions.py offers a wide range of features for building Python-powered Discord bots and web applications alike:
```

Author: This change is for SEO. The word python needs to stay. Happy to expand the usecase, but that word needs to stay

Author: Readability change

`interactions.py offers a wide range of features for building Python-powered Discord bots and web applications alike:`

- ✅ 100% coverage of the Discord API
- ✅ Dynamic cache with TTL support
- ✅ Modern and Pythonic API for easy interaction with Discord
- ✅ Proper rate-limit handling
- ✅ Feature parity with most other Discord API wrappers
- ✅ Fully automated command synchronisation

> In addition to its core functionality, Interactions.py also provides a range of optional extensions, allowing you to further customize your bot and add new features with ease.

Reviewer: ```suggestion
In addition to core functionality, interactions.py provides a range of optional extensions, allowing you to further customize your bot and add new features with ease.
```

## Extensibility

> So the stock library doesn't do what you want? No problem! Builtin extensions are available to extend the functionality of the library. And if none of those strike your fancy there are a myriad of other extension libraries available.

Reviewer: ```suggestion
So the base library doesn't do what you want? No problem! With builtin extensions, you are able to extend the functionality of the library. And if none of those pique your interest, there are a myriad of other extension libraries available.
```

Just type `bot.load("extension")`

<details>
    <summary>Extensions</summary>

   ### Prefixed Commands

   Prefixed commands, message commands, or legacy commands.
   Whatever you want to call them, by default the `interactions.py` library will not handle these. But rest assured this extension will get you going

  - ✅ Automatic command registration
  - ✅ Annotation support

> 

Reviewer: ```suggestion
```

based newline

  ### Debug Ext

  A fully featured debug and utilities suite to help you get your bots made

  ### Jurigged

  A hot reloading extension allowing you to automagically update your bot without reboots

  ### Sentry

  Integrates Sentry.io error tracking into your bot with a single line

</details>

## Where do I start?

> Getting started with Interactions.py is easy! Simply install it via `pip` and start building your Discord bot in Python:

Reviewer: ```suggestion
Getting started with interactions.py is easy! Simply install it via `pip` and start building your Discord application in Python:
```

`pip install -U discord-py-interactions`
```python
import interactions

bot = interactions.Client()

@interactions.listen()
async def on_start():
    print("Bot is ready!")

bot.start("token")
```

> With Interactions.py, you can quickly and easily build complex Discord bots with Python. Check out our [quickstart guide](https://interactionspy.rtfd.io/en/latest/quickstart.html) for more information.

Reviewer: ```suggestion
With interactions.py, you can quickly and easily build complex Discord applications with Python. Check out our [quickstart guide](https://interactionspy.rtfd.io/en/latest/quickstart.html) for more information.
```

## Key Features
Interactions.py offers a range of features for building Discord bots with Python, including:
- ✅ 100% coverage of the Discord API
- ✅ Dynamic cache with TTL support
- ✅ Modern and Pythonic API for easy interaction with Discord
- ✅ Proper rate-limit handling
- ✅ Feature parity with most other Discord API wrappers
- ✅ Fully automated command synchronisation

In addition to its core functionality, Interactions.py also provides a range of optional extensions, allowing you to further customize your bot and add new features with ease.

## Extensibility

So the stock library doesn't do what you want? No problem! Builtin extensions are available to extend the functionality of the library. And if none of those strike your fancy there are a myriad of other extension libraries available.

Just type `bot.load("extension")`

<details>
    <summary>Extensions</summary>

   ### Prefixed Commands

   Prefixed commands, message commands, or legacy commands.
   Whatever you want to call them, by default the `interactions.py` library will not handle these. But rest assured this extension will get you going

>   - ✅ Hybrid Commands

Reviewer 2: There are no hybrid commands as of right now.
